DirectXTex/DirectXTex.h

```diff
@@ -613,6 +613,10 @@ namespace DirectX
     constexpr unsigned long TEX_FILTER_MODE_MASK = 0xF00000;
     constexpr unsigned long TEX_FILTER_SRGB_MASK = 0xF000000;
 
+    // For reporting the image compression/conversion progress
```

Comment:
Callbacks should use ``std::function`` instead of legacy C style callbacks.

```
 _In_opt_ std::function<bool __cdecl(size_t, size_t)> statusCallBack = nullptr);
```

See the WIC functions for examples as well as [UVAtlas](https://github.com/microsoft/UVAtlas/blob/main/UVAtlas/inc/UVAtlas.h).

Reply:
A subtle but important detail: The public interface passes the ``std::function<>`` by value. All the internal functions take it by ``const std::function<>&``.

Reply:
Done.